apps/opencs/view/render/terrainshapemode.cpp

```diff
@@ -49,6 +50,12 @@ CSVRender::TerrainShapeMode::TerrainShapeMode (WorldspaceWidget *worldspaceWidge
 {
 }
 
+CSVRender::TerrainShapeMode::~TerrainShapeMode ()
+{
+    if (mBrushDraw)
+        mBrushDraw.reset();
```

Comment:
I don't understand the purpose of this code.

Reply:
I wanted to make sure that the pointer would be dropped, but now to think of it, isn't std::unique_ptr exactly the kind of smart pointer that will be destroyed automatically anyway?

Reply:
Exactly.